pykokkos/interface/type_inference.py

```diff
@@ -0,0 +1,279 @@
+            updated_types.inferred_types[param.name] = "bool"
+            updated_types.is_arg.add(param.name)
+
+    return updated_types
+
+
+def infer_other_args(
+    param_list: List[inspect.Parameter], 
+    policy_params: int,
+    args_list: List[Any],
+    start_idx: int,
+    updated_types: UpdatedTypes
+    ) -> UpdatedTypes:
+    '''
+    Infer the types of arguments (after the policy arguments)
+
+    param_list: list of parameter objects that are present in the workunit signature
+    policy_params: the number of initial parameters that are dedicated to policy (in param_list/signature)
+    args_list: List of arguments passed to the parallel dispactch (e.g args for parallal_for())
+    start_idx: The index for the first non policy argument in args_list
+    updated_types: UpdatedTypes object to store inferred types information
+    returns: Updated UpdatedTypes object with inferred types
+    '''
+
+    # DataType class has all supported pk datatypes, we ignore class members starting with __
+    supported_np_dtypes = [attr for attr in dir(DataType) if not attr.startswith("__")]
+
+    for i in range(policy_params , len(param_list)):
+        param = param_list[i]
+        value = args_list[start_idx + i - policy_params]
+
+        if isinstance(value, View) and value.layout != Layout.LayoutDefault:
+            updated_types.layout_change[param.name] = "LayoutRight" if value.layout == Layout.LayoutRight else "LayoutLeft"
+
+        if param.annotation is not inspect._empty:
+            continue
+
+        param_type = type(value).__name__
+
+        # switch integer values over 31 bits (signed positive value) to pk.int64
+        if param_type == "int" and value.bit_length() > 31:
+            param_type = "numpy:int64"
+
+        # check if package name is numpy (handling numpy primitives)
+        pckg_name = type(value).__module__
+
+        if pckg_name == "numpy":
+            if param_type not in supported_np_dtypes:
+                raise TypeError(f"Numpy type {param_type} is unsupported")
+
+            # numpy:<type>, Will switch to pk.<type> in parser.fix_types
+            param_type = pckg_name +":"+ param_type
+
+        if isinstance(value, View):
+            view_dtype = get_pk_datatype(value.dtype)
+            if not view_dtype:
+                raise TypeError("Cannot infer datatype for view:", param.name)
+
+            param_type = "View"+str(len(value.shape))+"D:"+view_dtype
+
+        updated_types.inferred_types[param.name] = param_type 
+        updated_types.is_arg.add(param.name)
+
+    return updated_types
+
+
+def get_pk_datatype(value):
+    '''
+    value: value whose datatype is to be determined
+    returns the type of custom pkDataType as string
+    '''
+
+    dtype = None
+    if isinstance(value, DataType):
+        dtype = str(value.name)
+
+    elif inspect.isclass(value) and issubclass(value, DataTypeClass):
+        dtype = str(value.__name__)
+
+    return dtype
```

Comment:
should we raise an exception or return `None` if no match

Reply:
It is raised after this function returns to the the caller